src/SignalR/server/Core/ref/Microsoft.AspNetCore.SignalR.Core.netcoreapp3.0.cs

```diff
@@ -167,6 +167,14 @@ public partial class HubConnectionStore
             public void Reset() { }
         }
     }
+    public abstract partial class HubDispatcher<THub> where THub : Microsoft.AspNetCore.SignalR.Hub
+    {
+        protected HubDispatcher() { }
+        public abstract System.Threading.Tasks.Task DispatchMessageAsync(Microsoft.AspNetCore.SignalR.HubConnectionContext connection, Microsoft.AspNetCore.SignalR.Protocol.HubMessage hubMessage);
+        public abstract System.Collections.Generic.IReadOnlyList<System.Type> GetParameterTypes(string name);
```

Comment:
This is super unfortunate as a public API @anurse. Can we file an issue to see if we can do anything tactical for 3.0?

Reply:
Why is a HubDispatcher registered with DI to begin with? Is it just so we don't have to construct in manually in HubConnectionHandler? It looks like we don't actually use anything other than the DefaultHubDispatcher even in our tests. There are no other HubConnectionHandler implementations at all, not even test mocks.

I even checked https://github.com/Azure/azure-signalr to see if they might have taken a dependency on it somehow, but it doesn't seem so. We should just construct a DefaultHubDispatcher manually in HubConnectionHandler.

Reply:
It was to provide a way for users to do something like what hub pipeline modules let you do in ASP.NET SignalR. I'm OK with removing the extensibility point for now and considering "real" pipeline modules (https://github.com/aspnet/AspNetCore/issues/5353)

Reply:
@mikaelm12 I think we're in agreement. For now, lets make `HubDispatcher<THub>` internal and remove it from DI. I think you just need manually construct a DefaultHubDispatcher in the HubConnectionHandler ctor.

Reply:
Seems fine. Also, we're doing hub pipeline modules next release.